### PR TITLE
Add gifencoder package

### PR DESCRIPTION
**Pull request by cprecioso**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/gifencoder/gifencoder-tests.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import GIFEncoder = require('gifencoder');
+import fs = require('fs');
+
+declare const pngFileStream: NodeJS.ReadableStream;
```
Comment (peterblazejewicz) [inline types/gifencoder/gifencoder-tests.ts:4]: just btw, these tests are just for linting and they are not executed really, 
so there is no need for a tricks like that one and there could be a proper code 
here that obtains a stream  - just like in the real life scenario.
It won't be executed, just run through compiler/linter checks

File: types/gifencoder/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": ["es6", "DOM"],
```
Comment (peterblazejewicz) [inline types/gifencoder/tsconfig.json:4]: Do you need a DOM reference to make it working? (for a `.d.ts` definition file that is)

File: types/gifencoder/gifencoder-tests.ts (hunk 1 above)
Comment (cprecioso) [reply types/gifencoder/gifencoder-tests.ts:4]: I know, but the original example depended on the `png-file-stream` package and so I just decided to replace that with a `declare const`. Should I change it?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
Obviously it covers most scenarios people can use based on README/examples, there is a more non-documented api in the class itself, but for regular usage that's OK.
@cprecioso thx for the contribution!

File: types/gifencoder/tsconfig.json (hunk 2 above)
Comment (cprecioso) [reply types/gifencoder/tsconfig.json:4]: Yes, the main method to add frames to the gif takes Canvas contexts. https://github.com/DefinitelyTyped/DefinitelyTyped/pull/47303/files#diff-82143f2d9bf49fd0be45ef359156961cR22

File: types/gifencoder/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// Type definitions for gifencoder 2.0
+// Project: https://github.com/eugeneware/gifencoder#readme
+// Definitions by: Carlos Precioso <https://github.com/DefinitelyTyped>
```
Comment (peterblazejewicz) [inline types/gifencoder/index.d.ts:3]:
```suggestion
// Definitions by: Carlos Precioso <https://github.com/cprecioso>
```

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
Please amend the owner of the DT package, thanks!

**Review by peterblazejewicz — APPROVED** [review]

**Comment by orta** [comment]
Cool, looks good to me 👍🏻

**Pull request merged**